psychopy/hardware/camera/__init__.py

```diff
@@ -579,7 +579,7 @@ def isSameDevice(self, other):
         elif isinstance(other, Camera):
             return getattr(other, "_capture", None) == self
         elif isinstance(other, dict) and "device" in other:
-            return other['device'] == self._device
+            return other['deviceName'] == self._device
```

Comment:
Wouldn't this mean that `getProfile` returns the first profile matching the device name rather than the specific profile (resolution and all) which we want?